### PR TITLE
[Merged by Bors] - chore: adaptations for nightly-2024-09-26

### PR DESCRIPTION
**Pull request by jcommelin**

### CODE REVIEW

File: Mathlib/Control/Basic.lean [hunk 1]
```diff
@@ -19,11 +19,8 @@ variable {α β γ : Type u}
 section Functor
 
 variable {f : Type u → Type v} [Functor f] [LawfulFunctor f]
```
Comment (Ruben-VandeVelde) [inline Mathlib/Control/Basic.lean:21]: No longer used, please remove

File: Mathlib/Control/Traversable/Instances.lean [hunk 2]
```diff
@@ -32,7 +32,7 @@ theorem Option.id_traverse {α} (x : Option α) : Option.traverse (pure : α →
 theorem Option.comp_traverse {α β γ} (f : β → F γ) (g : α → G β) (x : Option α) :
     Option.traverse (Comp.mk ∘ (f <$> ·) ∘ g) x =
       Comp.mk (Option.traverse f <$> Option.traverse g x) := by
-  cases x <;> simp! [functor_norm] <;> rfl
+  cases x <;> (simp! [functor_norm] <;> rfl)
```
Comment (Ruben-VandeVelde) [inline Mathlib/Control/Traversable/Instances.lean:35]: Huh. Curious why this is needed

**Review by Ruben-VandeVelde — APPROVED** [review]
Lgtm

File: Mathlib/Control/Traversable/Instances.lean (hunk 2 above)
Comment (kim-em) [reply Mathlib/Control/Traversable/Instances.lean:35]: One of the branches of `cases` is now closed by `simp` and doesn't need the `rfl`.

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**